scaccogatto/vue-waypoint
# refactor: move from vue-cli to vite

> With last Vue version there is an infinite callback calling in the library

> Divided PR in 2 commits to be easier to review

- Remove `.browserslistrc` https://vitejs.dev/guide/build.html#browser-compatibility
- Adjust Eslint a little, disable unneeded rules
- Update GitHub Actions node version to 14, most libraries deprecated 12
- Update GitHub Actions to run all checks
- Fix git hooks (husky) lint-staged, add commitlint
- Migrate from Jest to Vitest
- Build library in UMD and ESM format
- Fix App.vue v-for
- Simplify IntersectionObserver mock in tests
- Unwrap component.vue to component.ts because it had only script
- Infer some types, adjust main entry exports
- Simplify `cssHelpers` computation
- Fix scoped slots
- Run once prettier/eslint fix on the entire project. Except CHANGELOG.md

ps: Maybe a hook can also be extracted e.g.
```ts
const state = useWaypoint(elementRef, observerOptions)
```
ps3: Netlify fails because it uses a old distro image that should updated and it uses node 12 also

Closes #69

<img width="755" alt="image" src="https://user-images.githubusercontent.com/1881266/184256841-898630d8-c75f-44fe-a2e4-e063b902a24a.png">

<img width="764" alt="image" src="https://user-images.githubusercontent.com/1881266/184256826-003c63fb-d9f4-43a5-befb-9f804d56d764.png">
